src/components/containers/MovieList/MovieList.js

```diff
@@ -7,74 +7,69 @@ import AddMovieModal from '../../modals/AddMovieModal';
 
 const getMovieById = (id) => mockMoviesData.movies.find((movie) => movie.id === id);
 
-class MovieList extends React.Component {
-  constructor(props) {
-    super(props);
-    this.state = {
-      showEditForm: false,
-      showDeleteForm: false,
-      selectedMovieId: null,
-    };
-  }
+const MovieList = () => {
+  const [showEditForm, setShowEditForm] = useState(false);
+  const [showDeleteForm, setShowDeleteForm] = useState(false);
+  const [currentMovieId, setCurrentMovieId] = useState(null);
 
-  optionsHandler = (action, movieId) => {
-    this.setState({ selectedMovieId: movieId });
+  const optionsHandler = (action, movieId) => {
+    setCurrentMovieId(movieId);
     switch (action) {
       case 'edit':
-        this.setState({ showEditForm: true });
+        setShowEditForm(true);
         break;
       case 'delete':
-        this.setState({ showDeleteForm: true });
+        setShowDeleteForm(true);
         break;
       case 'close':
         break;
       default:
         throw new Error(`Unknown action ${action}`);
     }
-  }
+  };
 
-  onDeleteFormAction = (action, movieId) => {
+  const onDeleteFormAction = (action, movieId) => {
     switch (action) {
       case 'confirm':
         alert(`Movie ${movieId} to be deleted!`);
-        this.setState({ showDeleteForm: false, selectedMovieId: null });
+        setShowDeleteForm(false);
+        setCurrentMovieId(null);
         break;
       case 'close':
-        this.setState({ showDeleteForm: false, selectedMovieId: null });
+        setShowDeleteForm(false);
+        setCurrentMovieId(null);
         break;
       default:
         throw new Error(`Unknown action from Delete From: ${action}`);
     }
-  }
+  };
 
-  onEditFormAction = (action, movieId) => {
-    this.setState({ showEditForm: false, selectedMovieId: null });
-  }
+  const onEditFormAction = (action, movieId) => {
```

Comment:
why do you need `action` and `movieId` here?

Reply:
This wasn't implemented completely in Task-4 actually. The idea was that when you enter data in Edit form, you can initiate either 'Submit' or simple 'Close' (without saving) actions; `movieId` is the id of editing movie. But it seems that it will be refactored further anyway

Reply:
Refactored so far